### ISSUE TITLE
Update Groovy library from 2.4.7 to 2.4.12 or 2.4.13

### ISSUE THREAD

**Issue by homebeaver** [adempiere/adempiere#1548]
2.4.13 is the actual groovy-all.jar release from Nov, 2017 , see https://mvnrepository.com/artifact/org.codehaus.groovy/groovy-all

2.4.12 is inside eclipse NEON

**Comment by marcalwestf** [comment]
@homebeaver thanks for the hint.
It should be updated and tested.

Have you replaced the old Groovy library with the new one in the sources and tried it out?

**Comment by homebeaver** [comment]
Yes, I have replaced the groovy jar with 2.4.12 to test some enhencements which are written in groovy and are productive to our customers. It works fine with 2.4.12
For me groovy is an excellent way to make customer enhancements, because there is no need to compile the enhancement and bind it with the base AD. This is done dynamically. We use groovy in different ways

- we have groovy processes which are started from AD-menu, for expl we load payments from datev-OPOS [see here](https://klst-de.github.io/de-mpiere/3.datev.html)
- there are processes which are started on a button, exmpl: a bridge to excel or outlook to schedule an appointment
- we have callouts on fields 
- and we have workflow actions which are witten in groovy , see [here](http://wiki.idempiere.org/de/Workflow_(Fenster_ID-113)#Knotenaktion)

**Comment by homebeaver** [comment]
Oops, I closed it accidentally ...

**Comment by homebeaver** [comment]
This commit is in #1817 pull request
